code/game/objects/items/scanners.dm

```diff
@@ -265,28 +271,81 @@ REAGENT SCANNER
 		var/current_organ = list(
 			"name" = organ.name,
 			"status" = organ.organ_status == ORGAN_BRUISED ? "Bruised" : "Broken",
-			"damage" = organ.damage
+			"damage" = organ.damage,
+			"effects" = organ.damage_description,
 		)
 		damaged_organs += list(current_organ)
 	data["damaged_organs"] = damaged_organs
 
-	if(HAS_TRAIT(patient, TRAIT_UNDEFIBBABLE))
-		data["revivable_string"] = "Permanently deceased" // revivable_string is the actual information. "too much damage" etc.
-		data["revivable_boolean"] = FALSE // revivable_boolean is simply the TRUE/FALSE data entry used by tgui to color the revivable box
-	else if(HAS_TRAIT(patient, TRAIT_IMMEDIATE_DEFIB))
-		data["revivable_string"] = "Ready to reboot"
-		data["revivable_boolean"] = TRUE
+	var/organic_patient = !(patient.species.species_flags & (IS_SYNTHETIC|ROBOTIC_LIMBS))
+	var/revivable_patient = FALSE
+	
+	if(HAS_TRAIT(patient, TRAIT_IMMEDIATE_DEFIB))
+		revivable_patient = TRUE
+	else if(issynth(patient))
+		if(patient.health >= patient.get_death_threshold())
+			revivable_patient = TRUE
 	else if(patient.health + patient.getOxyLoss() + (DEFIBRILLATOR_HEALING_TIMES_SKILL(user.skills.getRating(SKILL_MEDICAL))) >= patient.get_death_threshold())
-		data["revivable_string"] = "Ready to defibrillate"
+		revivable_patient = TRUE
+
+	if(HAS_TRAIT(patient, TRAIT_UNDEFIBBABLE))
+		data["revivable_string"] = "Permanently deceased" // the actual information shown next to "revivable:" in tgui. "too much damage" etc.
+		data["revivable_boolean"] = FALSE // the actual TRUE/FALSE entry used by tgui. if false, revivable text is red. if true, revivable text is yellow
+	else if(organic_patient && !patient.has_working_organs())
+		data["revivable_string"] = "Not ready to defibrillate - heart too damaged"
+		data["revivable_boolean"] = FALSE
+	else if(revivable_patient)
+		data["revivable_string"] = "Ready to [organic_patient ? "defibrillate" : "reboot"]" // Ternary for defibrillate or reboot for some IC flavor
 		data["revivable_boolean"] = TRUE
 	else
-		data["revivable_string"] = "Not ready to defibrillate - repair damage first"
+		data["revivable_string"] = "Not ready to [organic_patient ? "defibrillate" : "reboot"] - repair damage above [patient.get_death_threshold() / patient.maxHealth * 100 - (DEFIBRILLATOR_HEALING_TIMES_SKILL(user.skills.getRating(SKILL_MEDICAL)))]%"
 		data["revivable_boolean"] = FALSE
 
 	// ADVICE
 	var/list/advice = list()
 	var/list/temp_advice = list()
 	if(!HAS_TRAIT(patient, TRAIT_UNDEFIBBABLE)) // only show advice at all if the patient is coming back
+		//random stuff that docs should be aware of. possible todo: make a system so we can put these in a collapsible tgui element if there's more added here.
+		if(patient.maxHealth != 100)
```

Comment:
this should really be a define if you're going to use it this way

Reply:
```dm
	/// Maximum health that should be possible.
	var/maxHealth = LIVING_DEFAULT_MAX_HEALTH
	/// Mob's current health
	var/health = LIVING_DEFAULT_HEALTH
```
```dm
	if(patient.maxHealth != LIVING_DEFAULT_MAX_HEALTH)
		advice += list(list(
			"advice" = "Patient has [patient.maxHealth / LIVING_DEFAULT_MAX_HEALTH * 100]% constitution.",
			"tooltip" = patient.maxHealth < LIVING_DEFAULT_MAX_HEALTH ? "Patient has less constitution than most humans." : "Patient has more constitution than most humans.",
			"icon" = patient.maxHealth < LIVING_DEFAULT_MAX_HEALTH ? "heart-broken" : "heartbeat",
			"color" = patient.maxHealth < LIVING_DEFAULT_MAX_HEALTH ? "grey" : "pink"
		))
```
### ?

Reply:
Ye